File: Leet_Code/Isomorphic_Strings.py
class Solution:
    def isIsomorphic(self, s: str, t: str) -> bool:
        iso_dict = {}
        for i in range(len(s)):
            if s[i] not in iso_dict and t[i] not in iso_dict.values():
                iso_dict[s[i]] = t[i]
            elif iso_dict.get(s[i]) != t[i]:
                return False
        return True

File: Leet_Code/Isomorphic_Strings.java
class Solution {
    public boolean isIsomorphic(String s, String t) {
        Map<Character, Character> letterMap1 = new HashMap<>();
        Map<Character, Character> letterMap2 = new HashMap<>();

        for (int i = 0; i < s.length(); i++) {
            Character c1 = s.charAt(i);
            Character c2 = t.charAt(i);
            if (letterMap1.containsKey(c1) && letterMap1.get(c1) != c2) {
                return false;
            } else if (letterMap2.containsKey(c2) && letterMap2.get(c2) != c1) {
                return false;
            } else {
                letterMap1.put(c1, c2);
                letterMap2.put(c2, c1);
            }
        }
        return true;
    }
}